google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineFacetEditor.java

```diff
@@ -172,6 +174,10 @@ public String getHelpTopic() {
   @Override
   public void onFacetInitialized(@NotNull Facet facet) {
     AppEngineWebIntegration.getInstance().setupDevServer();
+
+    UsageTrackerProvider.getInstance()
+        .trackEvent(GctTracking.APP_ENGINE_ADD_FACET)
```

Comment:
There are a few classes like the
com.intellij.facet.autodetecting.FacetDetector API that look like they
could be interesting.

There's also com.intellij.facet.ProjectWideFacetListener and
com.intellij.facet.FacetManager.Listener

no idea if they are what we're looking for but might be worth checking out.

On Tue, Aug 30, 2016 at 2:43 PM, João Martins <notifications@github.com>
wrote:

> In google-cloud-tools-plugin/src/com/google/cloud/tools/
> intellij/appengine/facet/AppEngineFacetEditor.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/850#discussion_r76855074>
> :
>
> > @@ -172,6 +174,10 @@ public String getHelpTopic() {
> >    @Override
> >    public void onFacetInitialized(@NotNull Facet facet) {
> >      AppEngineWebIntegration.getInstance().setupDevServer();
> > +
> > +    UsageTrackerProvider.getInstance()
> > +        .trackEvent(GctTracking.APP_ENGINE_ADD_FACET)
>
> An example I just found is when you start IJ and AppEngineFacet() is
> called, even though you aren't adding any facet (the IDE is constructing
> the project and constructing an AppEngineFacet). You'd be counting each
> time the browser is started as a new facet added.
>
> I'm afraid there are other cases where AppEngineFacet() is called but a
> facet isn't being added by a user that I'm not seeing.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/850/files/01bc1c342e5d6e94ffece143ffe84b37bc5f3738#r76855074>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HeYMinnq1xEPYpY4x62p4Q5gvG4oks5qlHm1gaJpZM4Jvw29>
> .
>